Basic Settings page created + Redesigned Buttons/Dialog

ProgDaniel opened the pull request:
A basic Settings Page has been created.

- Has a back button, both it and the OS back button returns the user to the Main page
- Has a delete account button; Currently no functionality.
- Does currently not have the temporary changes functionality for future settings.

Other changes:
- Question Dialogues has been updated to follow new visual standard
  - Changed colors; rounded borders
- CustomButtons have been updated to follow new visual standard
  - Changed colors; rounded borders

ProgDaniel committed:
WIP: Settings Menu

ProgDaniel committed:
WIP: Settings Page

ProgDaniel committed:
Settings screen & Question Dialog overhaul

The Settings Screen has been redesigned. Currently has the Delete Account button with no functionality

Question Dialog visual design has been updated.

ProgDaniel committed:
Question Dialog can take optional Text Color

Question Dialog can now be supplied an optional 'textColor' parameter that changes the color of the pop-up text

If ignored, uses the standard dark blue text color

ProgDaniel committed:
WIP: Question Dialog Mocking

LivHolm commented on lib/pages/settings/settings_page_bloc.dart:
```diff
@@ -0,0 +1,107 @@
+            }
+          }
+          break;
+          */
```
Find ud af om det her skal slettes.

LivHolm reviewed (changes requested):
Husk at lave unit test, på events og states.

Commit:
Added unit test functionality for Question Dialog

ProgDaniel committed:
Merge branch 'question_dialog_design' into settings_page

ProgDaniel committed:
Created Mock class for Question Dialog for testing

ProgDaniel committed:
Merge branch 'question_dialog_design' into settings_page

ProgDaniel committed:
WIP: Testing for Settings Page

ProgDaniel committed:
Various Settings Page changes + Tests

Added 'buttonPadding' value to values.dart, for button separation distance

Removed unused Card widget from Settings Page

Settings Page States Tests; Events Tests
(Bloc Test still WORK IN PROGRESS)

ProgDaniel committed:
Settings Cleanup + Removed folder in wrong branch

Removed Unused Libraries

Removed commented function for unsaved settings; may re-implement later

Removed work in progress history page from another branch

ProgDaniel committed:
Actually cleaned up Settings

Changes from previous commit were stashed; applied this time.

Commit:
Added new blocTestWidget environment

Commit:
Implemented bloc_test_widget.dart

LivHolm commented on test/mocks/question_dialog.dart:
```diff
@@ -0,0 +1,57 @@
+  }
+}
+
+/*
+
+void main() async {
+  TestWidgetsFlutterBinding.ensureInitialized();
+  await dotenv.load();
+
+  group("MainPage", () {
+    late MainPageBloc bloc;
+
+    setUp(() {
+      QuestionDialog.setInstance(MockQuestionDialog());
+      QuackService.init(MockQuackService());
+      SpotifyService.init(MockSpotifyService());
+      bloc = MainPageBloc(
+          context: MockBuildContext(), positionHelper: MockPositionHelper());
+    });
+
+    // blocTest<MainPageBloc, MainPageState>("ButtonPressed -> QuestionDialog",
+    //     build: () {
+    //       bloc.state
+    //           .copyWith(playerState: MockSpotifyService.getMockPlayerState());
+    //       return bloc;
+    //     },
+    //     act: (bloc) => bloc.add(
+    //         const ButtonPressed(buttonEvent: MainButtonEvent.refreshPlaylist)),
+    //     expect: () => []);
+  });
+}
+
+*/
```
Slet?

LivHolm reviewed (commented):
Test køre uden fejl og alt virker til at være implementeret fint ^^
